stytch/session/session_test.go

```diff
@@ -135,6 +139,103 @@ func TestAuthenticateJWTLocal(t *testing.T) {
 	})
 }
 
+func TestAuthenticateWithClaims(t *testing.T) {
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Handle the async JWKS fetch.
+		if strings.HasPrefix(r.URL.Path, "/sessions/jwks/") {
+			w.Write([]byte(`{"keys": []}`))
+			return
+		}
+
+		// This is the test request
+		if r.URL.Path == "/sessions/authenticate" {
+			// There are  many other fields in this response, but these are the only ones we need
+			// for this test.
+			w.Write([]byte(`{
+			  "session": {
+			    "expires_at": "2022-06-29T19:53:48Z",
+			    "last_accessed_at": "2022-06-29T17:54:13Z",
+			    "session_id": "session-test-aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
+			    "started_at": "2022-06-29T17:53:48Z",
+			    "user_id": "user-test-00000000-0000-0000-0000-000000000000",
+
+			    "custom_claims": {
+			      "https://my-app.example.net/custom-claim": {
+			        "number": 1,
+			        "array": [1, "foo", null],
+			        "nested": {
+			          "data": "here"
+			        }
+			      }
+			    }
+			  }
+			}`))
+			return
+		}
+
+		http.Error(w, "Bad Request", http.StatusBadRequest)
+		return
+	}))
+
+	client, err := stytchapi.NewAPIClient(
+		config.Env("anything"),
+		"project-test-00000000-0000-0000-0000-000000000000",
+		"secret-test-11111111-1111-1111-1111-111111111111",
+		stytchapi.WithBaseURI(srv.URL),
+	)
+	require.NoError(t, err)
+
+	req := &stytch.SessionsAuthenticateParams{
+		SessionToken: "fake session token",
+	}
+
+	{
+		var claims map[string]interface{}
+		_, err := client.Sessions.AuthenticateWithClaims(req, &claims)
+		require.NoError(t, err)
+
+		type object = map[string]interface{}
+		expected := object{
+			"https://my-app.example.net/custom-claim": object{
+				// Remember that numbers without specified types unmarshal as float64.
+				"number": float64(1),
+				"array":  []interface{}{float64(1), "foo", nil},
+				"nested": object{
+					"data": "here",
+				},
+			},
+		}
+		assert.Equal(t, expected, claims)
+	}
```

Comment:
(I know I wrote it, but now I don't like it lol) I think these should be subtests (`t.Run`) instead of blocks. It'd help us be clear that we're testing different unmarshal targets. It shouldn't hurt to share `srv` between both subtests.

Reply:
https://github.com/stytchauth/stytch-go/pull/77/commits/dfbb30b779e6f147f8826be224139e93d3315407